### PR TITLE
[Merged by Bors] - feat(Analysis/LocallyConvex/WithSeminorms): congr lemmas

### PR DESCRIPTION
**Pull request by ADedecker**
This adds `WithSeminorms.congr` which allows to replace a family of seminorm by an equivalent one. We use that to prove that one can always replace the family by a directed family (and a nice one if the indexing set is a `LocallyFiniteOrderBot`).

---

- [x] depends on: #5812

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5812
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Analysis/LocallyConvex/WithSeminorms.lean [hunk 1]
```diff
@@ -628,7 +627,7 @@ theorem continuous_iff_continuous_comp {q : SeminormFamily 𝕜₂ F ι'} [Topol
 #align seminorm.continuous_iff_continuous_comp Seminorm.continuous_iff_continuous_comp
 
 theorem continuous_from_bounded {p : SeminormFamily 𝕝 E ι} {q : SeminormFamily 𝕝₂ F ι'}
-    [TopologicalSpace E] (hp : WithSeminorms p) [TopologicalSpace F] (hq : WithSeminorms q)
+    {_ : TopologicalSpace E} (hp : WithSeminorms p) {_ : TopologicalSpace F} (hq : WithSeminorms q)
```
Comment (sgouezel) [inline Mathlib/Analysis/LocallyConvex/WithSeminorms.lean:630]: What's the reason for this change?

File: Mathlib/Analysis/LocallyConvex/WithSeminorms.lean [hunk 2]
```diff
@@ -661,6 +660,61 @@ theorem cont_normedSpace_to_withSeminorms (E) [SeminormedAddCommGroup E] [Normed
...
+section Congr
+
+namespace WithSeminorms
+
+variable [Nonempty ι] [Nonempty ι']
+variable [NormedField 𝕜] [AddCommGroup E] [Module 𝕜 E]
+variable [NormedField 𝕜₂] [AddCommGroup F] [Module 𝕜₂ F]
+variable {σ₁₂ : 𝕜 →+* 𝕜₂} [RingHomIsometric σ₁₂]
+
+protected theorem congr {p : SeminormFamily 𝕜 E ι} {q : SeminormFamily 𝕜 E ι'}
```
Comment (sgouezel) [inline Mathlib/Analysis/LocallyConvex/WithSeminorms.lean:672]: Maybe add a docstring to this one.

File: Mathlib/Analysis/LocallyConvex/WithSeminorms.lean (hunk 1 above)
Comment (ADedecker) [reply Mathlib/Analysis/LocallyConvex/WithSeminorms.lean:630]: With the old version [this line](https://github.com/leanprover-community/mathlib4/blob/92a117f9897a20ea6e9175824a031349ed375b30/Mathlib/Analysis/LocallyConvex/WithSeminorms.lean#L680) doesn't work. More generally, I feel like everything goes way smoother when things that can be inferred from various places (these topologies can either be inferred from the expected type or from the types of the arguments) aren't also searched by the typeclass system, especially since Lean4 seems to be even more strict on that than Lean3.

**Comment by sgouezel** [comment]
bors r+
Thanks!
